jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -104,11 +104,19 @@ object XGBoost extends Serializable {
     import DataUtils._
     val partitionedTrainingSet = repartitionData(trainingSet, numWorkers)
     val appName = partitionedTrainingSet.context.appName
+
+    val groupData: Seq[Seq[Int]] = xgBoostConfMap.get("groupData") match {
+      case Some(gpData) => gpData.asInstanceOf[Seq[Seq[Int]]]
+      case None => null
```

Comment:
because of your `optimization`, you will serialize a null and pass to executors when users are not doing ranking....why not just judge whether .get('xxxx') isDefined?